include/Core/Collision/PolygonalCollider.hpp

```diff
@@ -76,7 +89,12 @@ namespace obe::Collision
          * \param tagType List you want to clear (Tag / Accepted /Rejected)
          */
         void clearTags(ColliderTagType tagType);
-        [[nodiscard]] bool doesCollide(const Transform::UnitVector& offset) const;
+        /**
+         * \brief Checks if the collider is intersecting other colliders
+         * \param offset The offset to apply to the source collider
+         * \return std::vector of intersected colliders (empty if none found)
+         */
+        [[nodiscard]] std::vector<PolygonalCollider*> doesCollide(const Transform::UnitVector& offset) const;
         /**
```

Comment:
Why not returning a CollisionData here too ?

Reply:
Because `doesCollide` only check the colliders that can be touched in the offset, it doesn't return a new calculated offset (contrary to `getMaximumDistanceBeforeCollision`).

Reply:
You can always return a zero offset, it makes the API clearer and allows the user to benefit tof future methods of `CollisionData`.